tests/parser/test_shard_mask_parsing.py

```diff
@@ -0,0 +1,146 @@
+import unittest
+
+from lark import ParseError
+
+from fetchai.ledger.parser.etch_parser import EtchParser, UseWildcardShardMask, UnparsableAddress
+
+CONTRACT_TEXT = """
+persistent sharded balance : UInt64;
+persistent owner_name : String;
+
+@init
+function setup(owner : Address)
+  use balance[owner];
+  balance.set(owner, 1000000u64);
+endfunction
+
+@action
+function transfer(from: Address, to: Address, amount: UInt64)
+
+  use balance[from, to];
+
+  // Check if the sender has enough balance to proceed
+  if (balance.get(from) >= amount)
+
+    // update the account balances
+    balance.set(from, balance.get(from) - amount);
+    balance.set(to, balance.get(to, 0u64) + amount);
+  endif
+
+endfunction
+
+@query
+function balance(address: Address) : UInt64
+    use balance[address];
+    return balance.get(address, 0u64);
+endfunction
+
+"""
+
+# For testing use statements outside annotated functions
+# @init-setup() calls set_balance(), which uses a global outside an annotated function
+# @action-transfer() uses globals directly, and calls the sub(), which doesn't use globals
+NON_ENTRY_GLOBAL = """
+persistent sharded balance : UInt64;
+
+@init
+function setup(owner : Address)
+  set_balance(owner, 1000000u64);
+endfunction
+
+function set_balance(owner: Address, value: UInt64)
+    use balance[owner];
+    balance.set(owner, value);
+endfunction
+
+@action
+function transfer(from: Address, to: Address, amount: UInt64)
+
+  use balance[from, to];
```

Comment:
Would be good to add another test here to cover a case with a static.

```
use balance[from, to, "constant"];
```

For example

Reply:
Added, though could possibly be improved - I'm simply testing that known strings generate known hashes for now.